{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping 2bRAD with Bowtie"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "| Sample  | Weight (grams in shell)| Length (mm) | Raw 2bRAD reads |\n",
    "|---------|------------------------|-------------|-----------------|\n",
    "| hc1_2B  | 2.2                    |  17.41      | 1521\n",
    "| hc1_4B  | 1.9                    |  20.43           | 2294290\n",
    "| hc2_15B | 2.2                    |  25.33      | 2498\n",
    "| hc2_17  | 1.1                    |  19.38      | (seq 4x)best = 2964986\n",
    "| hc3_1   | 2.2                    |  26.79        | 1863108\n",
    "| hc3_5   | 1.9                    |  19.5       | 1915\n",
    "| hc3_7   | 1.4                    | 18.43            | 2438612\n",
    "| hc3_10  | 1.2                    |  19.80           | 4091281\n",
    "| hc3_11  | 2.1                    |   20.54          | 2901608\n",
    "| ss2_9B  | 1.5                    | 17.2        | 749\n",
    "| ss2_14B | 2.2                    | 21.02       | 29745\n",
    "| ss2_18B | 5.3                    | 35.76       | 3016\n",
    "| ss3_3B  | 2.1                    | 22.71       | 2076769\n",
    "| ss3_14B | 2.4                    | 24.71       | 2661152\n",
    "| ss3_15B | 2.6                    | 26.78       | 2816364\n",
    "| ss3_16B | 4.2                    | 39.57       | 2070520\n",
    "| ss3_20  | 2.3                    | 26.96       | 1656456\n",
    "| ss5_18  | 5.4                    | 27.34       | 2358172\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Users/sr320/git-repos/student-fish546-2016/jupyter'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1914464 reads; of these:\r\n",
      "  1914464 (100.00%) were unpaired; of these:\r\n",
      "    1910198 (99.78%) aligned 0 times\r\n",
      "    3432 (0.18%) aligned exactly 1 time\r\n",
      "    834 (0.04%) aligned >1 times\r\n",
      "0.22% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC1-1_TTAGGC-TCATG_L001_R1_001.fastq.gz \\\n",
    "-p 6 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC1-1.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC1-1.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC1-1.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC1-1.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/git-repos/student-fish546-2016/jupyter\n"
     ]
    }
   ],
   "source": [
    "cd /Users/sr320/git-repos/student-fish546-2016/jupyter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1521 reads; of these:\r\n",
      "  1521 (100.00%) were unpaired; of these:\r\n",
      "    1516 (99.67%) aligned 0 times\r\n",
      "    5 (0.33%) aligned exactly 1 time\r\n",
      "    0 (0.00%) aligned >1 times\r\n",
      "0.33% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC1-2_CAGATC-CGCTATG_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC1-2.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC1-2.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC1-2.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC1-2.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2294290 reads; of these:\r\n",
      "  2294290 (100.00%) were unpaired; of these:\r\n",
      "    2288709 (99.76%) aligned 0 times\r\n",
      "    4265 (0.19%) aligned exactly 1 time\r\n",
      "    1316 (0.06%) aligned >1 times\r\n",
      "0.24% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC1-4_GGCTAC-TCATG_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC1-4.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC1-4.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC1-4.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC1-4.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2498 reads; of these:\r\n",
      "  2498 (100.00%) were unpaired; of these:\r\n",
      "    2492 (99.76%) aligned 0 times\r\n",
      "    5 (0.20%) aligned exactly 1 time\r\n",
      "    1 (0.04%) aligned >1 times\r\n",
      "0.24% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC2-15_CGATGT-CGCTATG_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC2-15.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC2-15.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC2-15.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC2-15.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2964986 reads; of these:\r\n",
      "  2964986 (100.00%) were unpaired; of these:\r\n",
      "    2953231 (99.60%) aligned 0 times\r\n",
      "    6972 (0.24%) aligned exactly 1 time\r\n",
      "    4783 (0.16%) aligned >1 times\r\n",
      "0.40% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC2-17B_TTAGGC-ATGAGCAC_L002_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC2-17B.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC2-17B.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC2-17B.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC2-17B.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1863108 reads; of these:\r\n",
      "  1863108 (100.00%) were unpaired; of these:\r\n",
      "    1857869 (99.72%) aligned 0 times\r\n",
      "    3692 (0.20%) aligned exactly 1 time\r\n",
      "    1547 (0.08%) aligned >1 times\r\n",
      "0.28% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-1_TTAGGC-ATGCAT_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-1.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-1.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-1.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-1.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1915 reads; of these:\r\n",
      "  1915 (100.00%) were unpaired; of these:\r\n",
      "    1909 (99.69%) aligned 0 times\r\n",
      "    4 (0.21%) aligned exactly 1 time\r\n",
      "    2 (0.10%) aligned >1 times\r\n",
      "0.31% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-5_TAGCTT-CGCTATG_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-5.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-5.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-5.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-5.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2438612 reads; of these:\r\n",
      "  2438612 (100.00%) were unpaired; of these:\r\n",
      "    2431323 (99.70%) aligned 0 times\r\n",
      "    5047 (0.21%) aligned exactly 1 time\r\n",
      "    2242 (0.09%) aligned >1 times\r\n",
      "0.30% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-7_ACTTGA-TCAGCAC_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-7.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-7.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-7.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-7.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4091281 reads; of these:\n",
      "  4091281 (100.00%) were unpaired; of these:\n",
      "    4080550 (99.74%) aligned 0 times\n",
      "    7474 (0.18%) aligned exactly 1 time\n",
      "    3257 (0.08%) aligned >1 times\n",
      "0.26% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-10_GGCTAC-ATGCAT_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-10.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-10.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-10.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-10.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2901608 reads; of these:\r\n",
      "  2901608 (100.00%) were unpaired; of these:\r\n",
      "    2894358 (99.75%) aligned 0 times\r\n",
      "    5116 (0.18%) aligned exactly 1 time\r\n",
      "    2134 (0.07%) aligned >1 times\r\n",
      "0.25% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-11_TTAGGC-ATGAGCAC_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-fast \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-11.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-11.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-11.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-11.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2901608 reads; of these:\r\n",
      "  2901608 (100.00%) were unpaired; of these:\r\n",
      "    1872442 (64.53%) aligned 0 times\r\n",
      "    527038 (18.16%) aligned exactly 1 time\r\n",
      "    502128 (17.31%) aligned >1 times\r\n",
      "35.47% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015/HC3-11_TTAGGC-ATGAGCAC_L001_R1_001.fastq.gz \\\n",
    "-p 8 \\\n",
    "--very-sensitive-local \\\n",
    "-S /Volumes/caviar/wd/2016-11-12/HC3-11.sam\n",
    "!samtools view -bS /Volumes/caviar/wd/2016-11-12/HC3-11.sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12/HC3-11.bam\n",
    "!samtools index /Volumes/caviar/wd/2016-11-12/HC3-11.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web/nightingales/O_lurida/2bRAD_Dec2015\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/web/nightingales/O_lurida/2bRAD_Dec2015"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1853604 reads; of these:\n",
      "  1853604 (100.00%) were unpaired; of these:\n",
      "    1205281 (65.02%) aligned 0 times\n",
      "    329041 (17.75%) aligned exactly 1 time\n",
      "    319282 (17.22%) aligned >1 times\n",
      "34.98% overall alignment rate\n",
      "2685751 reads; of these:\n",
      "  2685751 (100.00%) were unpaired; of these:\n",
      "    1654380 (61.60%) aligned 0 times\n",
      "    501862 (18.69%) aligned exactly 1 time\n",
      "    529509 (19.72%) aligned >1 times\n",
      "38.40% overall alignment rate\n",
      "1817265 reads; of these:\n",
      "  1817265 (100.00%) were unpaired; of these:\n",
      "    1147629 (63.15%) aligned 0 times\n",
      "    333311 (18.34%) aligned exactly 1 time\n",
      "    336325 (18.51%) aligned >1 times\n",
      "36.85% overall alignment rate\n",
      "4101311 reads; of these:\n",
      "  4101311 (100.00%) were unpaired; of these:\n",
      "    2534706 (61.80%) aligned 0 times\n",
      "    745790 (18.18%) aligned exactly 1 time\n",
      "    820815 (20.01%) aligned >1 times\n",
      "38.20% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2473 reads; of these:\n",
      "  2473 (100.00%) were unpaired; of these:\n",
      "    1584 (64.05%) aligned 0 times\n",
      "    403 (16.30%) aligned exactly 1 time\n",
      "    486 (19.65%) aligned >1 times\n",
      "35.95% overall alignment rate\n",
      "2960842 reads; of these:\n",
      "  2960842 (100.00%) were unpaired; of these:\n",
      "    1854082 (62.62%) aligned 0 times\n",
      "    534232 (18.04%) aligned exactly 1 time\n",
      "    572528 (19.34%) aligned >1 times\n",
      "37.38% overall alignment rate\n",
      "5769741 reads; of these:\n",
      "  5769741 (100.00%) were unpaired; of these:\n",
      "    3548264 (61.50%) aligned 0 times\n",
      "    1052938 (18.25%) aligned exactly 1 time\n",
      "    1168539 (20.25%) aligned >1 times\n",
      "38.50% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2881295 reads; of these:\n",
      "  2881295 (100.00%) were unpaired; of these:\n",
      "    1781097 (61.82%) aligned 0 times\n",
      "    527483 (18.31%) aligned exactly 1 time\n",
      "    572715 (19.88%) aligned >1 times\n",
      "38.18% overall alignment rate\n",
      "3055296 reads; of these:\n",
      "  3055296 (100.00%) were unpaired; of these:\n",
      "    1969371 (64.46%) aligned 0 times\n",
      "    546528 (17.89%) aligned exactly 1 time\n",
      "    539397 (17.65%) aligned >1 times\n",
      "35.54% overall alignment rate\n",
      "1914464 reads; of these:\n",
      "  1914464 (100.00%) were unpaired; of these:\n",
      "    1258781 (65.75%) aligned 0 times\n",
      "    345235 (18.03%) aligned exactly 1 time\n",
      "    310448 (16.22%) aligned >1 times\n",
      "34.25% overall alignment rate\n",
      "1521 reads; of these:\n",
      "  1521 (100.00%) were unpaired; of these:\n",
      "    1000 (65.75%) aligned 0 times\n",
      "    268 (17.62%) aligned exactly 1 time\n",
      "    253 (16.63%) aligned >1 times\n",
      "34.25% overall alignment rate\n",
      "2257529 reads; of these:\n",
      "  2257529 (100.00%) were unpaired; of these:\n",
      "    1455410 (64.47%) aligned 0 times\n",
      "    402974 (17.85%) aligned exactly 1 time\n",
      "    399145 (17.68%) aligned >1 times\n",
      "35.53% overall alignment rate\n",
      "2294290 reads; of these:\n",
      "  2294290 (100.00%) were unpaired; of these:\n",
      "    1470984 (64.11%) aligned 0 times\n",
      "    423291 (18.45%) aligned exactly 1 time\n",
      "    400015 (17.44%) aligned >1 times\n",
      "35.89% overall alignment rate\n",
      "2973594 reads; of these:\n",
      "  2973594 (100.00%) were unpaired; of these:\n",
      "    1859970 (62.55%) aligned 0 times\n",
      "    519830 (17.48%) aligned exactly 1 time\n",
      "    593794 (19.97%) aligned >1 times\n",
      "37.45% overall alignment rate\n",
      "2869998 reads; of these:\n",
      "  2869998 (100.00%) were unpaired; of these:\n",
      "    1799183 (62.69%) aligned 0 times\n",
      "    532304 (18.55%) aligned exactly 1 time\n",
      "    538511 (18.76%) aligned >1 times\n",
      "37.31% overall alignment rate\n",
      "1807 reads; of these:\n",
      "  1807 (100.00%) were unpaired; of these:\n",
      "    1157 (64.03%) aligned 0 times\n",
      "    319 (17.65%) aligned exactly 1 time\n",
      "    331 (18.32%) aligned >1 times\n",
      "35.97% overall alignment rate\n",
      "4425665 reads; of these:\n",
      "  4425665 (100.00%) were unpaired; of these:\n",
      "    2850404 (64.41%) aligned 0 times\n",
      "    791637 (17.89%) aligned exactly 1 time\n",
      "    783624 (17.71%) aligned >1 times\n",
      "35.59% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "3081940 reads; of these:\n",
      "  3081940 (100.00%) were unpaired; of these:\n",
      "    1972626 (64.01%) aligned 0 times\n",
      "    550660 (17.87%) aligned exactly 1 time\n",
      "    558654 (18.13%) aligned >1 times\n",
      "35.99% overall alignment rate\n",
      "2974985 reads; of these:\n",
      "  2974985 (100.00%) were unpaired; of these:\n",
      "    1921418 (64.59%) aligned 0 times\n",
      "    534599 (17.97%) aligned exactly 1 time\n",
      "    518968 (17.44%) aligned >1 times\n",
      "35.41% overall alignment rate\n",
      "2498 reads; of these:\n",
      "  2498 (100.00%) were unpaired; of these:\n",
      "    1617 (64.73%) aligned 0 times\n",
      "    469 (18.78%) aligned exactly 1 time\n",
      "    412 (16.49%) aligned >1 times\n",
      "35.27% overall alignment rate\n",
      "3032101 reads; of these:\n",
      "  3032101 (100.00%) were unpaired; of these:\n",
      "    1970663 (64.99%) aligned 0 times\n",
      "    541831 (17.87%) aligned exactly 1 time\n",
      "    519607 (17.14%) aligned >1 times\n",
      "35.01% overall alignment rate\n",
      "2065139 reads; of these:\n",
      "  2065139 (100.00%) were unpaired; of these:\n",
      "    1330220 (64.41%) aligned 0 times\n",
      "    373908 (18.11%) aligned exactly 1 time\n",
      "    361011 (17.48%) aligned >1 times\n",
      "35.59% overall alignment rate\n",
      "2964986 reads; of these:\n",
      "  2964986 (100.00%) were unpaired; of these:\n",
      "    1834557 (61.87%) aligned 0 times\n",
      "    543196 (18.32%) aligned exactly 1 time\n",
      "    587233 (19.81%) aligned >1 times\n",
      "38.13% overall alignment rate\n",
      "3379 reads; of these:\n",
      "  3379 (100.00%) were unpaired; of these:\n",
      "    2295 (67.92%) aligned 0 times\n",
      "    587 (17.37%) aligned exactly 1 time\n",
      "    497 (14.71%) aligned >1 times\n",
      "32.08% overall alignment rate\n",
      "129 reads; of these:\n",
      "  129 (100.00%) were unpaired; of these:\n",
      "    80 (62.02%) aligned 0 times\n",
      "    26 (20.16%) aligned exactly 1 time\n",
      "    23 (17.83%) aligned >1 times\n",
      "37.98% overall alignment rate\n",
      "2484661 reads; of these:\n",
      "  2484661 (100.00%) were unpaired; of these:\n",
      "    1615231 (65.01%) aligned 0 times\n",
      "    447642 (18.02%) aligned exactly 1 time\n",
      "    421788 (16.98%) aligned >1 times\n",
      "34.99% overall alignment rate\n",
      "6279322 reads; of these:\n",
      "  6279322 (100.00%) were unpaired; of these:\n",
      "    4041952 (64.37%) aligned 0 times\n",
      "    1135415 (18.08%) aligned exactly 1 time\n",
      "    1101955 (17.55%) aligned >1 times\n",
      "35.63% overall alignment rate\n",
      "[bam_sort_core] merging from 3 files...\n",
      "2791198 reads; of these:\n",
      "  2791198 (100.00%) were unpaired; of these:\n",
      "    1727437 (61.89%) aligned 0 times\n",
      "    512655 (18.37%) aligned exactly 1 time\n",
      "    551106 (19.74%) aligned >1 times\n",
      "38.11% overall alignment rate\n",
      "2378462 reads; of these:\n",
      "  2378462 (100.00%) were unpaired; of these:\n",
      "    1533925 (64.49%) aligned 0 times\n",
      "    422810 (17.78%) aligned exactly 1 time\n",
      "    421727 (17.73%) aligned >1 times\n",
      "35.51% overall alignment rate\n",
      "3343053 reads; of these:\n",
      "  3343053 (100.00%) were unpaired; of these:\n",
      "    2106364 (63.01%) aligned 0 times\n",
      "    609758 (18.24%) aligned exactly 1 time\n",
      "    626931 (18.75%) aligned >1 times\n",
      "36.99% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2054031 reads; of these:\n",
      "  2054031 (100.00%) were unpaired; of these:\n",
      "    1288079 (62.71%) aligned 0 times\n",
      "    371929 (18.11%) aligned exactly 1 time\n",
      "    394023 (19.18%) aligned >1 times\n",
      "37.29% overall alignment rate\n",
      "3014825 reads; of these:\n",
      "  3014825 (100.00%) were unpaired; of these:\n",
      "    1900530 (63.04%) aligned 0 times\n",
      "    553772 (18.37%) aligned exactly 1 time\n",
      "    560523 (18.59%) aligned >1 times\n",
      "36.96% overall alignment rate\n",
      "64 reads; of these:\n",
      "  64 (100.00%) were unpaired; of these:\n",
      "    45 (70.31%) aligned 0 times\n",
      "    10 (15.62%) aligned exactly 1 time\n",
      "    9 (14.06%) aligned >1 times\n",
      "29.69% overall alignment rate\n",
      "3034 reads; of these:\n",
      "  3034 (100.00%) were unpaired; of these:\n",
      "    1912 (63.02%) aligned 0 times\n",
      "    578 (19.05%) aligned exactly 1 time\n",
      "    544 (17.93%) aligned >1 times\n",
      "36.98% overall alignment rate\n",
      "5333407 reads; of these:\n",
      "  5333407 (100.00%) were unpaired; of these:\n",
      "    3384333 (63.46%) aligned 0 times\n",
      "    975641 (18.29%) aligned exactly 1 time\n",
      "    973433 (18.25%) aligned >1 times\n",
      "36.54% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "1091951 reads; of these:\n",
      "  1091951 (100.00%) were unpaired; of these:\n",
      "    695326 (63.68%) aligned 0 times\n",
      "    200848 (18.39%) aligned exactly 1 time\n",
      "    195777 (17.93%) aligned >1 times\n",
      "36.32% overall alignment rate\n",
      "3256873 reads; of these:\n",
      "  3256873 (100.00%) were unpaired; of these:\n",
      "    2063281 (63.35%) aligned 0 times\n",
      "    595206 (18.28%) aligned exactly 1 time\n",
      "    598386 (18.37%) aligned >1 times\n",
      "36.65% overall alignment rate\n",
      "4091281 reads; of these:\n",
      "  4091281 (100.00%) were unpaired; of these:\n",
      "    2661452 (65.05%) aligned 0 times\n",
      "    729722 (17.84%) aligned exactly 1 time\n",
      "    700107 (17.11%) aligned >1 times\n",
      "34.95% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2901608 reads; of these:\n",
      "  2901608 (100.00%) were unpaired; of these:\n",
      "    1872442 (64.53%) aligned 0 times\n",
      "    527038 (18.16%) aligned exactly 1 time\n",
      "    502128 (17.31%) aligned >1 times\n",
      "35.47% overall alignment rate\n",
      "2529752 reads; of these:\n",
      "  2529752 (100.00%) were unpaired; of these:\n",
      "    1633686 (64.58%) aligned 0 times\n",
      "    455405 (18.00%) aligned exactly 1 time\n",
      "    440661 (17.42%) aligned >1 times\n",
      "35.42% overall alignment rate\n",
      "2437221 reads; of these:\n",
      "  2437221 (100.00%) were unpaired; of these:\n",
      "    1563868 (64.17%) aligned 0 times\n",
      "    447510 (18.36%) aligned exactly 1 time\n",
      "    425843 (17.47%) aligned >1 times\n",
      "35.83% overall alignment rate\n",
      "3103529 reads; of these:\n",
      "  3103529 (100.00%) were unpaired; of these:\n",
      "    1969229 (63.45%) aligned 0 times\n",
      "    563586 (18.16%) aligned exactly 1 time\n",
      "    570714 (18.39%) aligned >1 times\n",
      "36.55% overall alignment rate\n",
      "31711 reads; of these:\n",
      "  31711 (100.00%) were unpaired; of these:\n",
      "    20234 (63.81%) aligned 0 times\n",
      "    5608 (17.68%) aligned exactly 1 time\n",
      "    5869 (18.51%) aligned >1 times\n",
      "36.19% overall alignment rate\n",
      "2423148 reads; of these:\n",
      "  2423148 (100.00%) were unpaired; of these:\n",
      "    1536917 (63.43%) aligned 0 times\n",
      "    438888 (18.11%) aligned exactly 1 time\n",
      "    447343 (18.46%) aligned >1 times\n",
      "36.57% overall alignment rate\n",
      "1816827 reads; of these:\n",
      "  1816827 (100.00%) were unpaired; of these:\n",
      "    1178388 (64.86%) aligned 0 times\n",
      "    326123 (17.95%) aligned exactly 1 time\n",
      "    312316 (17.19%) aligned >1 times\n",
      "35.14% overall alignment rate\n",
      "2496994 reads; of these:\n",
      "  2496994 (100.00%) were unpaired; of these:\n",
      "    1613369 (64.61%) aligned 0 times\n",
      "    441702 (17.69%) aligned exactly 1 time\n",
      "    441923 (17.70%) aligned >1 times\n",
      "35.39% overall alignment rate\n",
      "1863108 reads; of these:\n",
      "  1863108 (100.00%) were unpaired; of these:\n",
      "    1206315 (64.75%) aligned 0 times\n",
      "    333410 (17.90%) aligned exactly 1 time\n",
      "    323383 (17.36%) aligned >1 times\n",
      "35.25% overall alignment rate\n",
      "2391482 reads; of these:\n",
      "  2391482 (100.00%) were unpaired; of these:\n",
      "    1542509 (64.50%) aligned 0 times\n",
      "    435811 (18.22%) aligned exactly 1 time\n",
      "    413162 (17.28%) aligned >1 times\n",
      "35.50% overall alignment rate\n",
      "2517012 reads; of these:\n",
      "  2517012 (100.00%) were unpaired; of these:\n",
      "    1635494 (64.98%) aligned 0 times\n",
      "    450940 (17.92%) aligned exactly 1 time\n",
      "    430578 (17.11%) aligned >1 times\n",
      "35.02% overall alignment rate\n",
      "3820533 reads; of these:\n",
      "  3820533 (100.00%) were unpaired; of these:\n",
      "    2457066 (64.31%) aligned 0 times\n",
      "    672610 (17.61%) aligned exactly 1 time\n",
      "    690857 (18.08%) aligned >1 times\n",
      "35.69% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "1915 reads; of these:\n",
      "  1915 (100.00%) were unpaired; of these:\n",
      "    1241 (64.80%) aligned 0 times\n",
      "    345 (18.02%) aligned exactly 1 time\n",
      "    329 (17.18%) aligned >1 times\n",
      "35.20% overall alignment rate\n",
      "2344151 reads; of these:\n",
      "  2344151 (100.00%) were unpaired; of these:\n",
      "    1527218 (65.15%) aligned 0 times\n",
      "    422350 (18.02%) aligned exactly 1 time\n",
      "    394583 (16.83%) aligned >1 times\n",
      "34.85% overall alignment rate\n",
      "2438612 reads; of these:\n",
      "  2438612 (100.00%) were unpaired; of these:\n",
      "    1589233 (65.17%) aligned 0 times\n",
      "    434479 (17.82%) aligned exactly 1 time\n",
      "    414900 (17.01%) aligned >1 times\n",
      "34.83% overall alignment rate\n",
      "2604877 reads; of these:\n",
      "  2604877 (100.00%) were unpaired; of these:\n",
      "    1677886 (64.41%) aligned 0 times\n",
      "    461338 (17.71%) aligned exactly 1 time\n",
      "    465653 (17.88%) aligned >1 times\n",
      "35.59% overall alignment rate\n",
      "3572907 reads; of these:\n",
      "  3572907 (100.00%) were unpaired; of these:\n",
      "    2277411 (63.74%) aligned 0 times\n",
      "    655049 (18.33%) aligned exactly 1 time\n",
      "    640447 (17.93%) aligned >1 times\n",
      "36.26% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "1830273 reads; of these:\n",
      "  1830273 (100.00%) were unpaired; of these:\n",
      "    1234395 (67.44%) aligned 0 times\n",
      "    327693 (17.90%) aligned exactly 1 time\n",
      "    268185 (14.65%) aligned >1 times\n",
      "32.56% overall alignment rate\n",
      "2549597 reads; of these:\n",
      "  2549597 (100.00%) were unpaired; of these:\n",
      "    1730531 (67.87%) aligned 0 times\n",
      "    442483 (17.36%) aligned exactly 1 time\n",
      "    376583 (14.77%) aligned >1 times\n",
      "32.13% overall alignment rate\n",
      "2432449 reads; of these:\n",
      "  2432449 (100.00%) were unpaired; of these:\n",
      "    1635931 (67.25%) aligned 0 times\n",
      "    431765 (17.75%) aligned exactly 1 time\n",
      "    364753 (15.00%) aligned >1 times\n",
      "32.75% overall alignment rate\n",
      "2132608 reads; of these:\n",
      "  2132608 (100.00%) were unpaired; of these:\n",
      "    1420420 (66.60%) aligned 0 times\n",
      "    380881 (17.86%) aligned exactly 1 time\n",
      "    331307 (15.54%) aligned >1 times\n",
      "33.40% overall alignment rate\n",
      "2721426 reads; of these:\n",
      "  2721426 (100.00%) were unpaired; of these:\n",
      "    1816309 (66.74%) aligned 0 times\n",
      "    486741 (17.89%) aligned exactly 1 time\n",
      "    418376 (15.37%) aligned >1 times\n",
      "33.26% overall alignment rate\n",
      "1989595 reads; of these:\n",
      "  1989595 (100.00%) were unpaired; of these:\n",
      "    1324440 (66.57%) aligned 0 times\n",
      "    357194 (17.95%) aligned exactly 1 time\n",
      "    307961 (15.48%) aligned >1 times\n",
      "33.43% overall alignment rate\n",
      "2611212 reads; of these:\n",
      "  2611212 (100.00%) were unpaired; of these:\n",
      "    1761158 (67.45%) aligned 0 times\n",
      "    457991 (17.54%) aligned exactly 1 time\n",
      "    392063 (15.01%) aligned >1 times\n",
      "32.55% overall alignment rate\n",
      "3279 reads; of these:\n",
      "  3279 (100.00%) were unpaired; of these:\n",
      "    2225 (67.86%) aligned 0 times\n",
      "    527 (16.07%) aligned exactly 1 time\n",
      "    527 (16.07%) aligned >1 times\n",
      "32.14% overall alignment rate\n",
      "2347454 reads; of these:\n",
      "  2347454 (100.00%) were unpaired; of these:\n",
      "    1575764 (67.13%) aligned 0 times\n",
      "    415670 (17.71%) aligned exactly 1 time\n",
      "    356020 (15.17%) aligned >1 times\n",
      "32.87% overall alignment rate\n",
      "2752 reads; of these:\n",
      "  2752 (100.00%) were unpaired; of these:\n",
      "    1888 (68.60%) aligned 0 times\n",
      "    487 (17.70%) aligned exactly 1 time\n",
      "    377 (13.70%) aligned >1 times\n",
      "31.40% overall alignment rate\n",
      "5693725 reads; of these:\n",
      "  5693725 (100.00%) were unpaired; of these:\n",
      "    3511347 (61.67%) aligned 0 times\n",
      "    1057659 (18.58%) aligned exactly 1 time\n",
      "    1124719 (19.75%) aligned >1 times\n",
      "38.33% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2550026 reads; of these:\n",
      "  2550026 (100.00%) were unpaired; of these:\n",
      "    1710369 (67.07%) aligned 0 times\n",
      "    455477 (17.86%) aligned exactly 1 time\n",
      "    384180 (15.07%) aligned >1 times\n",
      "32.93% overall alignment rate\n",
      "2378891 reads; of these:\n",
      "  2378891 (100.00%) were unpaired; of these:\n",
      "    1594193 (67.01%) aligned 0 times\n",
      "    416929 (17.53%) aligned exactly 1 time\n",
      "    367769 (15.46%) aligned >1 times\n",
      "32.99% overall alignment rate\n",
      "4823283 reads; of these:\n",
      "  4823283 (100.00%) were unpaired; of these:\n",
      "    3024219 (62.70%) aligned 0 times\n",
      "    872790 (18.10%) aligned exactly 1 time\n",
      "    926274 (19.20%) aligned >1 times\n",
      "37.30% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "7200858 reads; of these:\n",
      "  7200858 (100.00%) were unpaired; of these:\n",
      "    4482940 (62.26%) aligned 0 times\n",
      "    1321388 (18.35%) aligned exactly 1 time\n",
      "    1396530 (19.39%) aligned >1 times\n",
      "37.74% overall alignment rate\n",
      "[bam_sort_core] merging from 3 files...\n",
      "3166571 reads; of these:\n",
      "  3166571 (100.00%) were unpaired; of these:\n",
      "    1987268 (62.76%) aligned 0 times\n",
      "    571892 (18.06%) aligned exactly 1 time\n",
      "    607411 (19.18%) aligned >1 times\n",
      "37.24% overall alignment rate\n",
      "2410343 reads; of these:\n",
      "  2410343 (100.00%) were unpaired; of these:\n",
      "    1506473 (62.50%) aligned 0 times\n",
      "    437754 (18.16%) aligned exactly 1 time\n",
      "    466116 (19.34%) aligned >1 times\n",
      "37.50% overall alignment rate\n",
      "4644253 reads; of these:\n",
      "  4644253 (100.00%) were unpaired; of these:\n",
      "    2911429 (62.69%) aligned 0 times\n",
      "    823009 (17.72%) aligned exactly 1 time\n",
      "    909815 (19.59%) aligned >1 times\n",
      "37.31% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2883515 reads; of these:\n",
      "  2883515 (100.00%) were unpaired; of these:\n",
      "    1936370 (67.15%) aligned 0 times\n",
      "    511117 (17.73%) aligned exactly 1 time\n",
      "    436028 (15.12%) aligned >1 times\n",
      "32.85% overall alignment rate\n",
      "3727402 reads; of these:\n",
      "  3727402 (100.00%) were unpaired; of these:\n",
      "    2519557 (67.60%) aligned 0 times\n",
      "    656712 (17.62%) aligned exactly 1 time\n",
      "    551133 (14.79%) aligned >1 times\n",
      "32.40% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2779744 reads; of these:\n",
      "  2779744 (100.00%) were unpaired; of these:\n",
      "    1855319 (66.74%) aligned 0 times\n",
      "    498279 (17.93%) aligned exactly 1 time\n",
      "    426146 (15.33%) aligned >1 times\n",
      "33.26% overall alignment rate\n",
      "2152891 reads; of these:\n",
      "  2152891 (100.00%) were unpaired; of these:\n",
      "    1368752 (63.58%) aligned 0 times\n",
      "    404847 (18.80%) aligned exactly 1 time\n",
      "    379292 (17.62%) aligned >1 times\n",
      "36.42% overall alignment rate\n",
      "3397537 reads; of these:\n",
      "  3397537 (100.00%) were unpaired; of these:\n",
      "    2148502 (63.24%) aligned 0 times\n",
      "    620379 (18.26%) aligned exactly 1 time\n",
      "    628656 (18.50%) aligned >1 times\n",
      "36.76% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "3634398 reads; of these:\n",
      "  3634398 (100.00%) were unpaired; of these:\n",
      "    2313237 (63.65%) aligned 0 times\n",
      "    644106 (17.72%) aligned exactly 1 time\n",
      "    677055 (18.63%) aligned >1 times\n",
      "36.35% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "4969598 reads; of these:\n",
      "  4969598 (100.00%) were unpaired; of these:\n",
      "    3156253 (63.51%) aligned 0 times\n",
      "    896077 (18.03%) aligned exactly 1 time\n",
      "    917268 (18.46%) aligned >1 times\n",
      "36.49% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2895556 reads; of these:\n",
      "  2895556 (100.00%) were unpaired; of these:\n",
      "    1854553 (64.05%) aligned 0 times\n",
      "    511822 (17.68%) aligned exactly 1 time\n",
      "    529181 (18.28%) aligned >1 times\n",
      "35.95% overall alignment rate\n",
      "3177842 reads; of these:\n",
      "  3177842 (100.00%) were unpaired; of these:\n",
      "    2037291 (64.11%) aligned 0 times\n",
      "    574583 (18.08%) aligned exactly 1 time\n",
      "    565968 (17.81%) aligned >1 times\n",
      "35.89% overall alignment rate\n",
      "3384474 reads; of these:\n",
      "  3384474 (100.00%) were unpaired; of these:\n",
      "    2137290 (63.15%) aligned 0 times\n",
      "    616397 (18.21%) aligned exactly 1 time\n",
      "    630787 (18.64%) aligned >1 times\n",
      "36.85% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files...\n",
      "2840887 reads; of these:\n",
      "  2840887 (100.00%) were unpaired; of these:\n",
      "    1895289 (66.71%) aligned 0 times\n",
      "    502655 (17.69%) aligned exactly 1 time\n",
      "    442943 (15.59%) aligned >1 times\n",
      "33.29% overall alignment rate\n",
      "3713 reads; of these:\n",
      "  3713 (100.00%) were unpaired; of these:\n",
      "    2490 (67.06%) aligned 0 times\n",
      "    659 (17.75%) aligned exactly 1 time\n",
      "    564 (15.19%) aligned >1 times\n",
      "32.94% overall alignment rate\n",
      "1692 reads; of these:\n",
      "  1692 (100.00%) were unpaired; of these:\n",
      "    1067 (63.06%) aligned 0 times\n",
      "    317 (18.74%) aligned exactly 1 time\n",
      "    308 (18.20%) aligned >1 times\n",
      "36.94% overall alignment rate\n",
      "1540960 reads; of these:\n",
      "  1540960 (100.00%) were unpaired; of these:\n",
      "    991110 (64.32%) aligned 0 times\n",
      "    275688 (17.89%) aligned exactly 1 time\n",
      "    274162 (17.79%) aligned >1 times\n",
      "35.68% overall alignment rate\n",
      "2008378 reads; of these:\n",
      "  2008378 (100.00%) were unpaired; of these:\n",
      "    1262178 (62.85%) aligned 0 times\n",
      "    364987 (18.17%) aligned exactly 1 time\n",
      "    381213 (18.98%) aligned >1 times\n",
      "37.15% overall alignment rate\n",
      "1480148 reads; of these:\n",
      "  1480148 (100.00%) were unpaired; of these:\n",
      "    940256 (63.52%) aligned 0 times\n",
      "    270274 (18.26%) aligned exactly 1 time\n",
      "    269618 (18.22%) aligned >1 times\n",
      "36.48% overall alignment rate\n",
      "947549 reads; of these:\n",
      "  947549 (100.00%) were unpaired; of these:\n",
      "    609567 (64.33%) aligned 0 times\n",
      "    176687 (18.65%) aligned exactly 1 time\n",
      "    161295 (17.02%) aligned >1 times\n",
      "35.67% overall alignment rate\n",
      "1994893 reads; of these:\n",
      "  1994893 (100.00%) were unpaired; of these:\n",
      "    1280667 (64.20%) aligned 0 times\n",
      "    362117 (18.15%) aligned exactly 1 time\n",
      "    352109 (17.65%) aligned >1 times\n",
      "35.80% overall alignment rate\n",
      "3656 reads; of these:\n",
      "  3656 (100.00%) were unpaired; of these:\n",
      "    2328 (63.68%) aligned 0 times\n",
      "    662 (18.11%) aligned exactly 1 time\n",
      "    666 (18.22%) aligned >1 times\n",
      "36.32% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in HC*\n",
    "do\n",
    "/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Users/sr320/git-repos/student-fish546-2016/data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U $f \\\n",
    "-p 8 \\\n",
    "--very-sensitive-local  \\\n",
    "-S /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).sam\n",
    "samtools view -bS /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).bam\n",
    "samtools index /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).bam\n",
    "2> /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).err\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "for f in SS*\n",
    "do\n",
    "/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Users/sr320/git-repos/student-fish546-2016/data/Ostrea_lurida-Scaff-10k-bowtie-index \\\n",
    "-U $f \\\n",
    "-p 8 \\\n",
    "--very-sensitive-local  \\\n",
    "-S /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).sam\n",
    "2> /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).err\n",
    "samtools view -bS /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).sam \\\n",
    "| samtools sort -o /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).bam\n",
    "samtools index /Volumes/caviar/wd/2016-11-12b/$(basename \"${f}\" _R1_001.fastq.gz).bam\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
