{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Downloading Blast"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`ftp://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Index_of__blast_executables_LATEST_and_course-btea_01-Local_BLAST_ipynb_at_master_·_sr320_course-btea_1DA2BEC9.png\" alt=\"Index_of__blast_executables_LATEST_and_course-btea_01-Local_BLAST_ipynb_at_master_·_sr320_course-btea_1DA2BEC9.png\"/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Applications/bioinfo\n"
     ]
    }
   ],
   "source": [
    "cd /Applications/bioinfo/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "x ncbi-blast-2.5.0+/\n",
      "x ncbi-blast-2.5.0+/LICENSE\n",
      "x ncbi-blast-2.5.0+/README\n",
      "x ncbi-blast-2.5.0+/ChangeLog\n",
      "x ncbi-blast-2.5.0+/ncbi_package_info\n",
      "x ncbi-blast-2.5.0+/doc/\n",
      "x ncbi-blast-2.5.0+/doc/README.txt\n",
      "x ncbi-blast-2.5.0+/bin/\n",
      "x ncbi-blast-2.5.0+/bin/segmasker\n",
      "x ncbi-blast-2.5.0+/bin/blastdbcheck\n",
      "x ncbi-blast-2.5.0+/bin/psiblast\n",
      "x ncbi-blast-2.5.0+/bin/tblastn\n",
      "x ncbi-blast-2.5.0+/bin/blastn\n",
      "x ncbi-blast-2.5.0+/bin/makeblastdb\n",
      "x ncbi-blast-2.5.0+/bin/rpstblastn\n",
      "x ncbi-blast-2.5.0+/bin/legacy_blast.pl\n",
      "x ncbi-blast-2.5.0+/bin/windowmasker\n",
      "x ncbi-blast-2.5.0+/bin/dustmasker\n",
      "x ncbi-blast-2.5.0+/bin/makembindex\n",
      "x ncbi-blast-2.5.0+/bin/blastx\n",
      "x ncbi-blast-2.5.0+/bin/blastdb_aliastool\n",
      "x ncbi-blast-2.5.0+/bin/tblastx\n",
      "x ncbi-blast-2.5.0+/bin/makeprofiledb\n",
      "x ncbi-blast-2.5.0+/bin/blast_formatter\n",
      "x ncbi-blast-2.5.0+/bin/update_blastdb.pl\n",
      "x ncbi-blast-2.5.0+/bin/blastp\n",
      "x ncbi-blast-2.5.0+/bin/convert2blastmask\n",
      "x ncbi-blast-2.5.0+/bin/rpsblast\n",
      "x ncbi-blast-2.5.0+/bin/deltablast\n",
      "x ncbi-blast-2.5.0+/bin/blastdbcmd\n"
     ]
    }
   ],
   "source": [
    "!tar -xzvf ncbi-blast-2.5.0+-x64-macosx.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Applications/bioinfo/ncbi-blast-2.5.0+\n"
     ]
    }
   ],
   "source": [
    "cd ncbi-blast-2.5.0+/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ChangeLog          README             \u001b[34mdoc\u001b[m\u001b[m/\r\n",
      "LICENSE            \u001b[34mbin\u001b[m\u001b[m/               ncbi_package_info\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mblast_formatter\u001b[m\u001b[m*   \u001b[31mblastx\u001b[m\u001b[m*            \u001b[31mmakembindex\u001b[m\u001b[m*       \u001b[31mtblastn\u001b[m\u001b[m*\r\n",
      "\u001b[31mblastdb_aliastool\u001b[m\u001b[m* \u001b[31mconvert2blastmask\u001b[m\u001b[m* \u001b[31mmakeprofiledb\u001b[m\u001b[m*     \u001b[31mtblastx\u001b[m\u001b[m*\r\n",
      "\u001b[31mblastdbcheck\u001b[m\u001b[m*      \u001b[31mdeltablast\u001b[m\u001b[m*        \u001b[31mpsiblast\u001b[m\u001b[m*          \u001b[31mupdate_blastdb.pl\u001b[m\u001b[m*\r\n",
      "\u001b[31mblastdbcmd\u001b[m\u001b[m*        \u001b[31mdustmasker\u001b[m\u001b[m*        \u001b[31mrpsblast\u001b[m\u001b[m*          \u001b[31mwindowmasker\u001b[m\u001b[m*\r\n",
      "\u001b[31mblastn\u001b[m\u001b[m*            \u001b[31mlegacy_blast.pl\u001b[m\u001b[m*   \u001b[31mrpstblastn\u001b[m\u001b[m*\r\n",
      "\u001b[31mblastp\u001b[m\u001b[m*            \u001b[31mmakeblastdb\u001b[m\u001b[m*       \u001b[31msegmasker\u001b[m\u001b[m*\r\n"
     ]
    }
   ],
   "source": [
    "ls bin/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\r\n",
      "  blastx [-h] [-help] [-import_search_strategy filename]\r\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\r\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\r\n",
      "    [-negative_gilist filename] [-entrez_query entrez_query]\r\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\r\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\r\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\r\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\r\n",
      "    [-qcov_hsp_perc float_value] [-max_hsps int_value]\r\n",
      "    [-xdrop_ungap float_value] [-xdrop_gap float_value]\r\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value]\r\n",
      "    [-sum_stats bool_value] [-max_intron_length length] [-seg SEG_options]\r\n",
      "    [-soft_masking soft_masking] [-matrix matrix_name]\r\n",
      "    [-threshold float_value] [-culling_limit int_value]\r\n",
      "    [-best_hit_overhang float_value] [-best_hit_score_edge float_value]\r\n",
      "    [-window_size int_value] [-ungapped] [-lcase_masking] [-query_loc range]\r\n",
      "    [-strand strand] [-parse_deflines] [-query_gencode int_value]\r\n",
      "    [-outfmt format] [-show_gis] [-num_descriptions int_value]\r\n",
      "    [-num_alignments int_value] [-line_length line_length] [-html]\r\n",
      "    [-max_target_seqs num_sequences] [-num_threads int_value] [-remote]\r\n",
      "    [-comp_based_stats compo] [-use_sw_tback] [-version]\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "   Translated Query-Protein Subject BLAST 2.5.0+\r\n",
      "\r\n",
      "Use '-help' to print detailed descriptions of command line arguments\r\n"
     ]
    }
   ],
   "source": [
    "! bin/blastx -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\r\n",
      "  blastx [-h] [-help] [-import_search_strategy filename]\r\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\r\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\r\n",
      "    [-negative_gilist filename] [-entrez_query entrez_query]\r\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\r\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\r\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\r\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\r\n",
      "    [-qcov_hsp_perc float_value] [-max_hsps int_value]\r\n",
      "    [-xdrop_ungap float_value] [-xdrop_gap float_value]\r\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value]\r\n",
      "    [-sum_stats bool_value] [-max_intron_length length] [-seg SEG_options]\r\n",
      "    [-soft_masking soft_masking] [-matrix matrix_name]\r\n",
      "    [-threshold float_value] [-culling_limit int_value]\r\n",
      "    [-best_hit_overhang float_value] [-best_hit_score_edge float_value]\r\n",
      "    [-window_size int_value] [-ungapped] [-lcase_masking] [-query_loc range]\r\n",
      "    [-strand strand] [-parse_deflines] [-query_gencode int_value]\r\n",
      "    [-outfmt format] [-show_gis] [-num_descriptions int_value]\r\n",
      "    [-num_alignments int_value] [-line_length line_length] [-html]\r\n",
      "    [-max_target_seqs num_sequences] [-num_threads int_value] [-remote]\r\n",
      "    [-comp_based_stats compo] [-use_sw_tback] [-version]\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "   Translated Query-Protein Subject BLAST 2.5.0+\r\n",
      "\r\n",
      "OPTIONAL ARGUMENTS\r\n",
      " -h\r\n",
      "   Print USAGE and DESCRIPTION;  ignore all other parameters\r\n",
      " -help\r\n",
      "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\r\n",
      " -version\r\n",
      "   Print version number;  ignore other arguments\r\n",
      "\r\n",
      " *** Input query options\r\n",
      " -query <File_In>\r\n",
      "   Input file name\r\n",
      "   Default = `-'\r\n",
      " -query_loc <String>\r\n",
      "   Location on the query sequence in 1-based offsets (Format: start-stop)\r\n",
      " -strand <String, `both', `minus', `plus'>\r\n",
      "   Query strand(s) to search against database/subject\r\n",
      "   Default = `both'\r\n",
      " -query_gencode <Integer, values between: 1-6, 9-16, 21-25>\r\n",
      "   Genetic code to use to translate query (see user manual for details)\r\n",
      "   Default = `1'\r\n",
      "\r\n",
      " *** General search options\r\n",
      " -task <String, Permissible values: 'blastx' 'blastx-fast' >\r\n",
      "   Task to execute\r\n",
      "   Default = `blastx'\r\n",
      " -db <String>\r\n",
      "   BLAST database name\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -out <File_Out>\r\n",
      "   Output file name\r\n",
      "   Default = `-'\r\n",
      " -evalue <Real>\r\n",
      "   Expectation value (E) threshold for saving hits \r\n",
      "   Default = `10'\r\n",
      " -word_size <Integer, >=2>\r\n",
      "   Word size for wordfinder algorithm\r\n",
      " -gapopen <Integer>\r\n",
      "   Cost to open a gap\r\n",
      " -gapextend <Integer>\r\n",
      "   Cost to extend a gap\r\n",
      " -max_intron_length <Integer, >=0>\r\n",
      "   Length of the largest intron allowed in a translated nucleotide sequence\r\n",
      "   when linking multiple distinct alignments\r\n",
      "   Default = `0'\r\n",
      " -matrix <String>\r\n",
      "   Scoring matrix name (normally BLOSUM62)\r\n",
      " -threshold <Real, >=0>\r\n",
      "   Minimum word score such that the word is added to the BLAST lookup table\r\n",
      " -comp_based_stats <String>\r\n",
      "   Use composition-based statistics:\r\n",
      "       D or d: default (equivalent to 2 )\r\n",
      "       0 or F or f: No composition-based statistics\r\n",
      "       1: Composition-based statistics as in NAR 29:2994-3005, 2001\r\n",
      "       2 or T or t : Composition-based score adjustment as in Bioinformatics\r\n",
      "   21:902-911,\r\n",
      "       2005, conditioned on sequence properties\r\n",
      "       3: Composition-based score adjustment as in Bioinformatics 21:902-911,\r\n",
      "       2005, unconditionally\r\n",
      "   Default = `2'\r\n",
      "\r\n",
      " *** BLAST-2-Sequences options\r\n",
      " -subject <File_In>\r\n",
      "   Subject sequence(s) to search\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask\r\n",
      " -subject_loc <String>\r\n",
      "   Location on the subject sequence in 1-based offsets (Format: start-stop)\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask, remote\r\n",
      "\r\n",
      " *** Formatting options\r\n",
      " -outfmt <String>\r\n",
      "   alignment view options:\r\n",
      "     0 = Pairwise,\r\n",
      "     1 = Query-anchored showing identities,\r\n",
      "     2 = Query-anchored no identities,\r\n",
      "     3 = Flat query-anchored showing identities,\r\n",
      "     4 = Flat query-anchored no identities,\r\n",
      "     5 = BLAST XML,\r\n",
      "     6 = Tabular,\r\n",
      "     7 = Tabular with comment lines,\r\n",
      "     8 = Seqalign (Text ASN.1),\r\n",
      "     9 = Seqalign (Binary ASN.1),\r\n",
      "    10 = Comma-separated values,\r\n",
      "    11 = BLAST archive (ASN.1),\r\n",
      "    12 = Seqalign (JSON),\r\n",
      "    13 = Multiple-file BLAST JSON,\r\n",
      "    14 = Multiple-file BLAST XML2,\r\n",
      "    15 = Single-file BLAST JSON,\r\n",
      "    16 = Single-file BLAST XML2,\r\n",
      "    18 = Organism Report\r\n",
      "   \r\n",
      "   Options 6, 7 and 10 can be additionally configured to produce\r\n",
      "   a custom format specified by space delimited format specifiers.\r\n",
      "   The supported format specifiers are:\r\n",
      "   \t    qseqid means Query Seq-id\r\n",
      "   \t       qgi means Query GI\r\n",
      "   \t      qacc means Query accesion\r\n",
      "   \t   qaccver means Query accesion.version\r\n",
      "   \t      qlen means Query sequence length\r\n",
      "   \t    sseqid means Subject Seq-id\r\n",
      "   \t sallseqid means All subject Seq-id(s), separated by a ';'\r\n",
      "   \t       sgi means Subject GI\r\n",
      "   \t    sallgi means All subject GIs\r\n",
      "   \t      sacc means Subject accession\r\n",
      "   \t   saccver means Subject accession.version\r\n",
      "   \t   sallacc means All subject accessions\r\n",
      "   \t      slen means Subject sequence length\r\n",
      "   \t    qstart means Start of alignment in query\r\n",
      "   \t      qend means End of alignment in query\r\n",
      "   \t    sstart means Start of alignment in subject\r\n",
      "   \t      send means End of alignment in subject\r\n",
      "   \t      qseq means Aligned part of query sequence\r\n",
      "   \t      sseq means Aligned part of subject sequence\r\n",
      "   \t    evalue means Expect value\r\n",
      "   \t  bitscore means Bit score\r\n",
      "   \t     score means Raw score\r\n",
      "   \t    length means Alignment length\r\n",
      "   \t    pident means Percentage of identical matches\r\n",
      "   \t    nident means Number of identical matches\r\n",
      "   \t  mismatch means Number of mismatches\r\n",
      "   \t  positive means Number of positive-scoring matches\r\n",
      "   \t   gapopen means Number of gap openings\r\n",
      "   \t      gaps means Total number of gaps\r\n",
      "   \t      ppos means Percentage of positive-scoring matches\r\n",
      "   \t    frames means Query and subject frames separated by a '/'\r\n",
      "   \t    qframe means Query frame\r\n",
      "   \t    sframe means Subject frame\r\n",
      "   \t      btop means Blast traceback operations (BTOP)\r\n",
      "   \t    staxid means Subject Taxonomy ID\r\n",
      "   \t  ssciname means Subject Scientific Name\r\n",
      "   \t  scomname means Subject Common Name\r\n",
      "   \tsblastname means Subject Blast Name\r\n",
      "   \t sskingdom means Subject Super Kingdom\r\n",
      "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\r\n",
      "   \t\t\t (in numerical order)\r\n",
      "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\r\n",
      "   \t scomnames means unique Subject Common Name(s), separated by a ';'\r\n",
      "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order)\r\n",
      "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order) \r\n",
      "   \t    stitle means Subject Title\r\n",
      "   \tsalltitles means All Subject Title(s), separated by a '<>'\r\n",
      "   \t   sstrand means Subject Strand\r\n",
      "   \t     qcovs means Query Coverage Per Subject\r\n",
      "   \t   qcovhsp means Query Coverage Per HSP\r\n",
      "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\r\n",
      "   When not provided, the default value is:\r\n",
      "   'qacc sacc pident length mismatch gapopen qstart qend sstart send evalue\r\n",
      "   bitscore', which is equivalent to the keyword 'std'\r\n",
      "   Default = `0'\r\n",
      " -show_gis\r\n",
      "   Show NCBI GIs in deflines?\r\n",
      " -num_descriptions <Integer, >=0>\r\n",
      "   Number of database sequences to show one-line descriptions for\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -num_alignments <Integer, >=0>\r\n",
      "   Number of database sequences to show alignments for\r\n",
      "   Default = `250'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -line_length <Integer, >=1>\r\n",
      "   Line length for formatting alignments\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `60'\r\n",
      " -html\r\n",
      "   Produce HTML output?\r\n",
      "\r\n",
      " *** Query filtering options\r\n",
      " -seg <String>\r\n",
      "   Filter query sequence with SEG (Format: 'yes', 'window locut hicut', or\r\n",
      "   'no' to disable)\r\n",
      "   Default = `12 2.2 2.5'\r\n",
      " -soft_masking <Boolean>\r\n",
      "   Apply filtering locations as soft masks\r\n",
      "   Default = `false'\r\n",
      " -lcase_masking\r\n",
      "   Use lower case filtering in query and subject sequence(s)?\r\n",
      "\r\n",
      " *** Restrict search or results\r\n",
      " -gilist <String>\r\n",
      "   Restrict search of database to list of GI's\r\n",
      "    * Incompatible with:  negative_gilist, seqidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -seqidlist <String>\r\n",
      "   Restrict search of database to list of SeqId's\r\n",
      "    * Incompatible with:  gilist, negative_gilist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_gilist <String>\r\n",
      "   Restrict search of database to everything except the listed GIs\r\n",
      "    * Incompatible with:  gilist, seqidlist, remote, subject, subject_loc\r\n",
      " -entrez_query <String>\r\n",
      "   Restrict search with the given Entrez query\r\n",
      "    * Requires:  remote\r\n",
      " -db_soft_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as soft masking\r\n",
      "    * Incompatible with:  db_hard_mask, subject, subject_loc\r\n",
      " -db_hard_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as hard masking\r\n",
      "    * Incompatible with:  db_soft_mask, subject, subject_loc\r\n",
      " -qcov_hsp_perc <Real, 0..100>\r\n",
      "   Percent query coverage per hsp\r\n",
      " -max_hsps <Integer, >=1>\r\n",
      "   Set maximum number of HSPs per subject sequence to save for each query\r\n",
      " -culling_limit <Integer, >=0>\r\n",
      "   If the query range of a hit is enveloped by that of at least this many\r\n",
      "   higher-scoring hits, delete the hit\r\n",
      "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\r\n",
      " -best_hit_overhang <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm overhang value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -best_hit_score_edge <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm score edge value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -max_target_seqs <Integer, >=1>\r\n",
      "   Maximum number of aligned sequences to keep \r\n",
      "   Not applicable for outfmt <= 4\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  num_descriptions, num_alignments\r\n",
      "\r\n",
      " *** Statistical options\r\n",
      " -dbsize <Int8>\r\n",
      "   Effective length of the database \r\n",
      " -searchsp <Int8, >=0>\r\n",
      "   Effective length of the search space\r\n",
      " -sum_stats <Boolean>\r\n",
      "   Use sum statistics\r\n",
      "\r\n",
      " *** Search strategy options\r\n",
      " -import_search_strategy <File_In>\r\n",
      "   Search strategy to use\r\n",
      "    * Incompatible with:  export_search_strategy\r\n",
      " -export_search_strategy <File_Out>\r\n",
      "   File name to record the search strategy used\r\n",
      "    * Incompatible with:  import_search_strategy\r\n",
      "\r\n",
      " *** Extension options\r\n",
      " -xdrop_ungap <Real>\r\n",
      "   X-dropoff value (in bits) for ungapped extensions\r\n",
      " -xdrop_gap <Real>\r\n",
      "   X-dropoff value (in bits) for preliminary gapped extensions\r\n",
      " -xdrop_gap_final <Real>\r\n",
      "   X-dropoff value (in bits) for final gapped alignment\r\n",
      " -window_size <Integer, >=0>\r\n",
      "   Multiple hits window size, use 0 to specify 1-hit algorithm\r\n",
      " -ungapped\r\n",
      "   Perform ungapped alignment only?\r\n",
      "\r\n",
      " *** Miscellaneous options\r\n",
      " -parse_deflines\r\n",
      "   Should the query and subject defline(s) be parsed?\r\n",
      " -num_threads <Integer, >=1>\r\n",
      "   Number of threads (CPUs) to use in the BLAST search\r\n",
      "   Default = `1'\r\n",
      "    * Incompatible with:  remote\r\n",
      " -remote\r\n",
      "   Execute search remotely?\r\n",
      "    * Incompatible with:  gilist, seqidlist, negative_gilist, subject_loc,\r\n",
      "   num_threads\r\n",
      " -use_sw_tback\r\n",
      "   Compute locally optimal Smith-Waterman alignments?\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "#full path \n",
    "!/Applications/bioinfo/ncbi-blast-2.5.0\\+/bin/blastx -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Magic Blast?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\r\n",
      "  magicblast [-h] [-help] [-db database_name] [-gilist filename]\r\n",
      "    [-seqidlist filename] [-negative_gilist filename]\r\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\r\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\r\n",
      "    [-out output_file] [-gzo] [-word_size int_value] [-gapopen open_penalty]\r\n",
      "    [-gapextend extend_penalty] [-perc_identity float_value]\r\n",
      "    [-penalty penalty] [-lcase_masking] [-validate_seqs TF] [-infmt format]\r\n",
      "    [-paired] [-query_mate infile] [-sra accession] [-parse_deflines TF]\r\n",
      "    [-outfmt format] [-num_threads int_value] [-max_intron_length length]\r\n",
      "    [-score num] [-splice TF] [-reftype type] [-limit_lookup TF]\r\n",
      "    [-lookup_stride num] [-batch_size num] [-version]\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "   Short read mapper\r\n",
      "\r\n",
      "OPTIONAL ARGUMENTS\r\n",
      " -h\r\n",
      "   Print USAGE and DESCRIPTION;  ignore all other parameters\r\n",
      " -help\r\n",
      "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\r\n",
      " -version\r\n",
      "   Print version number;  ignore other arguments\r\n",
      "\r\n",
      " *** Input query options\r\n",
      " -query <File_In>\r\n",
      "   Input file name\r\n",
      "   Default = `-'\r\n",
      "    * Incompatible with:  sra\r\n",
      " -infmt <String, `asn1', `asn1b', `fasta', `fastc', `fastq'>\r\n",
      "   Input format for sequences\r\n",
      "   Default = `fasta'\r\n",
      "    * Incompatible with:  sra\r\n",
      " -paired\r\n",
      "   Input query sequences are paired\r\n",
      " -query_mate <File_In>\r\n",
      "   FASTA file with mates for query sequences (if given in another file)\r\n",
      "    * Requires:  query\r\n",
      " -sra <String>\r\n",
      "   Comma-separated SRA accessions\r\n",
      "    * Incompatible with:  query, infmt\r\n",
      "\r\n",
      " *** General search options\r\n",
      " -db <String>\r\n",
      "   BLAST database name\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -out <File_Out>\r\n",
      "   Output file name\r\n",
      "   Default = `-'\r\n",
      " -gzo\r\n",
      "   Output will be compressed\r\n",
      " -word_size <Integer, >=12>\r\n",
      "   Word size for wordfinder algorithm (length of best perfect match)\r\n",
      " -gapopen <Integer>\r\n",
      "   Cost to open a gap\r\n",
      " -gapextend <Integer>\r\n",
      "   Cost to extend a gap\r\n",
      " -penalty <Integer, <=0>\r\n",
      "   Penalty for a nucleotide mismatch\r\n",
      " -max_intron_length <Integer, >=0>\r\n",
      "   Length of the largest intron allowed in a translated nucleotide sequence\r\n",
      "   when linking multiple distinct alignments\r\n",
      "   Default = `2000'\r\n",
      "\r\n",
      " *** BLAST-2-Sequences options\r\n",
      " -subject <File_In>\r\n",
      "   Subject sequence(s) to search\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask\r\n",
      " -subject_loc <String>\r\n",
      "   Location on the subject sequence in 1-based offsets (Format: start-stop)\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask, remote\r\n",
      "\r\n",
      " *** Formatting options\r\n",
      " -outfmt <String, Permissible values: 'asn' 'sam' 'tabular' >\r\n",
      "   alignment view options:\r\n",
      "   sam = SAM format,\r\n",
      "   tabular = Tabular format,\r\n",
      "   text ASN.1\r\n",
      "   Default = `sam'\r\n",
      "\r\n",
      " *** Query filtering options\r\n",
      " -lcase_masking\r\n",
      "   Use lower case filtering in subject sequence(s)?\r\n",
      " -validate_seqs <Boolean>\r\n",
      "   Reject low quality sequences \r\n",
      "   Default = `true'\r\n",
      " -limit_lookup <Boolean>\r\n",
      "   Remove word seeds with high frequency in the searched database\r\n",
      " -lookup_stride <Integer>\r\n",
      "   Number of words to skip after collecting one while creating a lookup table\r\n",
      "   Default = `0'\r\n",
      "\r\n",
      " *** Restrict search or results\r\n",
      " -gilist <String>\r\n",
      "   Restrict search of database to list of GI's\r\n",
      "    * Incompatible with:  negative_gilist, seqidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -seqidlist <String>\r\n",
      "   Restrict search of database to list of SeqId's\r\n",
      "    * Incompatible with:  gilist, negative_gilist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_gilist <String>\r\n",
      "   Restrict search of database to everything except the listed GIs\r\n",
      "    * Incompatible with:  gilist, seqidlist, remote, subject, subject_loc\r\n",
      " -db_soft_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as soft masking\r\n",
      "    * Incompatible with:  db_hard_mask, subject, subject_loc\r\n",
      " -db_hard_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as hard masking\r\n",
      "    * Incompatible with:  db_soft_mask, subject, subject_loc\r\n",
      " -perc_identity <Real, 0..100>\r\n",
      "   Percent identity\r\n",
      "\r\n",
      " *** Miscellaneous options\r\n",
      " -parse_deflines <Boolean>\r\n",
      "   Should the query and subject defline(s) be parsed?\r\n",
      "   Default = `true'\r\n",
      " -num_threads <Integer, >=1>\r\n",
      "   Number of threads (CPUs) to use in the BLAST search\r\n",
      "   Default = `1'\r\n",
      "    * Incompatible with:  remote\r\n",
      "\r\n",
      " *** Mapping options\r\n",
      " -score <Integer>\r\n",
      "   Cutoff score for accepting a single non-spliced alignment\r\n",
      " -splice <Boolean>\r\n",
      "   Search for spliced alignments\r\n",
      " -reftype <String, `genome', `transcriptome'>\r\n",
      "   Type of the reference: genome or transcriptome\r\n",
      "   Default = `genome'\r\n",
      " -batch_size <Integer>\r\n",
      "   Number of query sequences read in a single batch\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/ncbi-magicblast-1.0.0/bin/magicblast -help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "Magic-BLAST is a tool for mapping large next-generation RNA or DNA sequencing\n",
    "runs against a whole genome or transcriptome. Each alignment optimizes\n",
    "a composite score, taking into account simultaneously the two reads of\n",
    "a pair, and in case of RNA-seq, locating the candidate introns and adding\n",
    "up the score of all exons. This is very different from other versions of\n",
    "BLAST, where each exon is scored as a separate hit and read-pairing is\n",
    "ignored.\n",
    "\n",
    "Magic-Blast incorporates within the NCBI BLAST code framework ideas\n",
    "developed in the NCBI Magic pipeline, in particular hit extensions by\n",
    "local walk and jump, which is faster than Smith-Waterman extension\n",
    "(http://www.ncbi.nlm.nih.gov/pubmed/26109056), and recursive clipping of\n",
    "mismatches near the edges of the reads, which avoids accumulating\n",
    "artefactual mismatches near splice sites and is needed to distinguish\n",
    "short indels from substitutions near the edges.\n",
    "\n",
    "We call the whole next generation run (from Illumina, Roche-454, ABI, or\n",
    "another sequencing platform excluding SOLiD), a query. The input reads may\n",
    "be provided as SRA accession or file in a SRA, FASTA, FASTQ, or FASTC format.\n",
    "Read pairs can be presented as parallel files, or as successive reads in a\n",
    "single file.\n",
    "\n",
    "The reference genome or transcriptome can be given as a BLAST database\n",
    "or a FASTA file. It is preferable to use BLAST database for large genomes,\n",
    "such as human, or transcript collections, such as all of RefSeq, Ensembl,\n",
    "or AceView. The procedure for creating a BLAST database is described below.\n",
    "\n",
    "The full list of options is listed when you use -help option.\n",
    "\n",
    "===================\n",
    "\n",
    "EXAMPLES:\n",
    "Use this command line to map RNA-seq reads in FASTA format to a reference\n",
    "genome in FASTA format:\n",
    "magicblast -query reads.fa -subject genome.fa\n",
    "\n",
    "Use this command line to map RNA-seq reads to a reference genome provided\n",
    "as a BLAST database:\n",
    "magicblast -query reads.fa -db genome\n",
    "\n",
    "Use this command line to map RNA-seq reads in FASTQ format:\n",
    "magicblast -query reads.fastq -db genome -infmt fastq\n",
    "\n",
    "The reads can also be provided from the standard input:\n",
    "cat reads.fa | magicblast -query - -db genome\n",
    "cat reads.fa | magicblast -db genome\n",
    "\n",
    "Magic-Blast recognizes read and reference file names with extension \".gz\"\n",
    "as compressed and decompresses them automatically using gzip.\n",
    "\n",
    "Use this command line to map SRA run to a genome:\n",
    "magicblast -sra sra_accession -db genome\n",
    "\n",
    "\n",
    "\n",
    "PAIRED-READS:\n",
    "\n",
    "For paired reads presented as successive entries in a single FASTA or FASTQ\n",
    "file, i.e. read 1 and 2 of fragment 1, then read 1 and 2 of fragment 2,\n",
    "etc., simply add the parameter '-paired':\n",
    "magicblast -query reads.fa -db genome -paired\n",
    "\n",
    "If you are using fastq-dump\n",
    "(http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&f=fastq-dump)\n",
    "to download NCBI SRA data or convert SRA file to the FASTQ format, use these\n",
    "parameters for runs with paired reads to create a single merged input file\n",
    "for Magic-Blast:\n",
    "fastq-dump -I --split-spot <accession or file>\n",
    "\n",
    "For paired reads presented in two parallel files, use these options:\n",
    "magicblast -query reads.fa -query_mate mates.fa -db genome\n",
    "\n",
    "\n",
    "RNA versus DNA:\n",
    "\n",
    "By default, Magic-Blast aligns RNA reads to a genome and reports spliced\n",
    "alignments, possibly spanning several exons. To disable spliced alignments,\n",
    "use the '-splice F' option. Use the '-reftype transcriptome' option, to\n",
    "map reads to a transcriptome database. These are example command lines:\n",
    "magicblast -query reads.fa -db genome -splice F\n",
    "magicblast -query reads.fa -db genome -reftype transcriptome\n",
    "\n",
    "Magic-Blast finds alignments between a read and a genome based on initial\n",
    "common word in both. To make mapping faster we first count word occurrences\n",
    "in the genome and disregard those that occur too often. With the\n",
    "'-reftype transcripts' option the words that are frequent in the transcript\n",
    "database are used for alignment.\n",
    "\n",
    "\n",
    "MULTI-THREADING\n",
    "\n",
    "To use multiple CPUs, specify the maximal number of threads with the\n",
    "'-num_threads' parameter:\n",
    "magicblast -query reads.fa -db genome -num_threads 10\n",
    "\n",
    "Currently this option works only with BLAST databases that contain more\n",
    "than one sequence.\n",
    "\n",
    "\n",
    "OUTPUT:\n",
    "\n",
    "By default, results are provided to the standard output in the SAM format.\n",
    "Use '-out filename' option to redirect output to a file.\n",
    "Use '-outfmt' option to specify the output format:\n",
    "-outfmt SAM : SAM format (default)\n",
    "-outfmt tabular : exports a simple tab delimited format defined below.\n",
    "\n",
    "The output can be also compressed, using the '-gzo' flag:\n",
    "magicblast -query reads.fa -db genome -out output.gz -gzo\n",
    "\n",
    "\n",
    "CREATION OF A BLAST DATABASE:\n",
    "\n",
    "Use this command line to create a BLAST database:\n",
    "makeblastdb -in fasta_file -dbtype nucl -parse_seqids -out database_name\n",
    "-title \"Database title\"\n",
    "\n",
    "The -parse_seqids option is required to keep the original sequence\n",
    "identifiers. Otherwise makeblastdb will generate its own identifiers.\n",
    "\n",
    "\n",
    "TABULAR OUTPUT FORMAT\n",
    "\n",
    "The tabular output format shows one alignment per line with these tab\n",
    "delimited fields:\n",
    "1. Query/read sequence identifier\n",
    "2. Reference sequence identifier\n",
    "3. Percent identity of the alignment\n",
    "4. Not used\n",
    "5. Not used\n",
    "6. Not used\n",
    "7. Alignment start position on the query sequence\n",
    "8. Alignment stop position on the query sequence\n",
    "9. Alignment start position on the reference sequence\n",
    "10. Alignment stop position on the reference sequence\n",
    "11. Not used\n",
    "12. Not used\n",
    "13. Alignment score\n",
    "14. Query strand\n",
    "15. Reference sequence strand\n",
    "16. Query/read length\n",
    "17. Alignment as extended BTOP string\n",
    "    This is the same BTOP string as in BLAST tabular output with a\n",
    "    few extensions:\n",
    "    - a number represents this many matches,\n",
    "    - two bases represent a mismatch and show query and reference base,\n",
    "    - base and gap or gap and base, show a gap in query or reference,\n",
    "    - ^<number>^ represents an intron of this number of bases,\n",
    "    - _<number>_ represents a gap in query of this number of bases,\n",
    "    - (<number>) shows number of query bases that are shared between\n",
    "    two parts of a spliced alignment; used when proper splice sites\n",
    "    were not found\n",
    "18. Number of different alignments reported for this query sequence\n",
    "19. Information about detected splice sites, polyA tails, or adapter\n",
    "    sequences found at the edges of the alignment\n",
    "20. Compartment - a unique identifier for all alignments that belong to\n",
    "    a single fragment. These can be two alignments for a pair of reads\n",
    "    or alignments to exons that were not spliced.\n",
    "21. Reverse complemented unaligned query sequence from the beginning\n",
    "    of the query, or '-' if the query aligns to the left edge\n",
    "22. Unaligned sequence at the end of the query, or '-'\n",
    "23. Reference sequence identifier where the mate is aligned, if\n",
    "    different from the identifier in column 2, otherwise '-'\n",
    "24. Alignment start position on the reference sequence for the mate, or\n",
    "    '-' if no alignment for the mate was found; a negative number\n",
    "    denotes a divergent pair\n",
    "25. Composite alignment score for all exons that belong to the fragment\n",
    "\n",
    "Thank you for testing this code and providing us with feedback. Please,\n",
    "let us know of any desired option, problem or difficulty.\n",
    "\n",
    "E-mail boratyng@ncbi.nlm.nih.gov or blast-help@ncbi.nlm.nih.gov with\n",
    "questions or comments.\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
