{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using Cyverse to map reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Discovery_Environment___CyVerse_1DA44227.png\" alt=\"Discovery_Environment___CyVerse_1DA44227.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# local"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\r\n",
      "Usage: \r\n",
      "bowtie [options]* <ebwt> {-1 <m1> -2 <m2> | --12 <r> | <s>} [<hit>]\r\n",
      "\r\n",
      "  <m1>    Comma-separated list of files containing upstream mates (or the\r\n",
      "          sequences themselves, if -c is set) paired with mates in <m2>\r\n",
      "  <m2>    Comma-separated list of files containing downstream mates (or the\r\n",
      "          sequences themselves if -c is set) paired with mates in <m1>\r\n",
      "  <r>     Comma-separated list of files containing Crossbow-style reads.  Can be\r\n",
      "          a mixture of paired and unpaired.  Specify \"-\" for stdin.\r\n",
      "  <s>     Comma-separated list of files containing unpaired reads, or the\r\n",
      "          sequences themselves, if -c is set.  Specify \"-\" for stdin.\r\n",
      "  <hit>   File to write hits to (default: stdout)\r\n",
      "Input:\r\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\r\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\r\n",
      "  -r                 query input files are raw one-sequence-per-line\r\n",
      "  -c                 query sequences given on cmd line (as <mates>, <singles>)\r\n",
      "  -C                 reads and index are in colorspace\r\n",
      "  -Q/--quals <file>  QV file(s) corresponding to CSFASTA inputs; use with -f -C\r\n",
      "  --Q1/--Q2 <file>   same as -Q, but for mate files 1 and 2 respectively\r\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input\r\n",
      "  -u/--qupto <int>   stop after first <int> reads/pairs (excl. skipped reads)\r\n",
      "  -5/--trim5 <int>   trim <int> bases from 5' (left) end of reads\r\n",
      "  -3/--trim3 <int>   trim <int> bases from 3' (right) end of reads\r\n",
      "  --phred33-quals    input quals are Phred+33 (default)\r\n",
      "  --phred64-quals    input quals are Phred+64 (same as --solexa1.3-quals)\r\n",
      "  --solexa-quals     input quals are from GA Pipeline ver. < 1.3\r\n",
      "  --solexa1.3-quals  input quals are from GA Pipeline ver. >= 1.3\r\n",
      "  --integer-quals    qualities are given as space-separated integers (not ASCII)\r\n",
      "  --large-index      force usage of a 'large' index, even if a small one is present\r\n",
      "Alignment:\r\n",
      "  -v <int>           report end-to-end hits w/ <=v mismatches; ignore qualities\r\n",
      "    or\r\n",
      "  -n/--seedmms <int> max mismatches in seed (can be 0-3, default: -n 2)\r\n",
      "  -e/--maqerr <int>  max sum of mismatch quals across alignment for -n (def: 70)\r\n",
      "  -l/--seedlen <int> seed length for -n (default: 28)\r\n",
      "  --nomaqround       disable Maq-like quality rounding for -n (nearest 10 <= 30)\r\n",
      "  -I/--minins <int>  minimum insert size for paired-end alignment (default: 0)\r\n",
      "  -X/--maxins <int>  maximum insert size for paired-end alignment (default: 250)\r\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (default: --fr)\r\n",
      "  --nofw/--norc      do not align to forward/reverse-complement reference strand\r\n",
      "  --maxbts <int>     max # backtracks for -n 2/3 (default: 125, 800 for --best)\r\n",
      "  --pairtries <int>  max # attempts to find mate for anchor hit (default: 100)\r\n",
      "  -y/--tryhard       try hard to find valid alignments, at the expense of speed\r\n",
      "  --chunkmbs <int>   max megabytes of RAM for best-first search frames (def: 64)\r\n",
      "Reporting:\r\n",
      "  -k <int>           report up to <int> good alignments per read (default: 1)\r\n",
      "  -a/--all           report all alignments per read (much slower than low -k)\r\n",
      "  -m <int>           suppress all alignments if > <int> exist (def: no limit)\r\n",
      "  -M <int>           like -m, but reports 1 random hit (MAPQ=0); requires --best\r\n",
      "  --best             hits guaranteed best stratum; ties broken by quality\r\n",
      "  --strata           hits in sub-optimal strata aren't reported (requires --best)\r\n",
      "Output:\r\n",
      "  -t/--time          print wall-clock time taken by search phases\r\n",
      "  -B/--offbase <int> leftmost ref offset = <int> in bowtie output (default: 0)\r\n",
      "  --quiet            print nothing but the alignments\r\n",
      "  --refout           write alignments to files refXXXXX.map, 1 map per reference\r\n",
      "  --refidx           refer to ref. seqs by 0-based index rather than name\r\n",
      "  --al <fname>       write aligned reads/pairs to file(s) <fname>\r\n",
      "  --un <fname>       write unaligned reads/pairs to file(s) <fname>\r\n",
      "  --max <fname>      write reads/pairs over -m limit to file(s) <fname>\r\n",
      "  --suppress <cols>  suppresses given columns (comma-delim'ed) in default output\r\n",
      "  --fullref          write entire ref name (default: only up to 1st space)\r\n",
      "Colorspace:\r\n",
      "  --snpphred <int>   Phred penalty for SNP when decoding colorspace (def: 30)\r\n",
      "     or\r\n",
      "  --snpfrac <dec>    approx. fraction of SNP bases (e.g. 0.001); sets --snpphred\r\n",
      "  --col-cseq         print aligned colorspace seqs as colors, not decoded bases\r\n",
      "  --col-cqual        print original colorspace quals, not decoded quals\r\n",
      "  --col-keepends     keep nucleotides at extreme ends of decoded alignment\r\n",
      "SAM:\r\n",
      "  -S/--sam           write hits in SAM format\r\n",
      "  --mapq <int>       default mapping quality (MAPQ) to print for SAM alignments\r\n",
      "  --sam-nohead       supppress header lines (starting with @) for SAM output\r\n",
      "  --sam-nosq         supppress @SQ header lines for SAM output\r\n",
      "  --sam-RG <text>    add <text> (usually \"lab=value\") to @RG line of SAM header\r\n",
      "Performance:\r\n",
      "  -o/--offrate <int> override offrate of index; must be >= index's offrate\r\n",
      "  -p/--threads <int> number of alignment threads to launch (default: 1)\r\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\r\n",
      "  --shmem            use shared mem for index; many 'bowtie's can share\r\n",
      "Other:\r\n",
      "  --seed <int>       seed for random number generator\r\n",
      "  --verbose          verbose output (for debugging)\r\n",
      "  --version          print version information and quit\r\n",
      "  -h/--help          print this usage message\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bowtie-1.1.2/bowtie"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bowtie 2 version 2.2.9 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\r\n",
      "Usage: \r\n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r>} [-S <sam>]\r\n",
      "\r\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\r\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\r\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <r>        Files with unpaired reads.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <sam>      File for SAM output (default: stdout)\r\n",
      "\r\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\r\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\r\n",
      "\r\n",
      "Options (defaults in parentheses):\r\n",
      "\r\n",
      " Input:\r\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\r\n",
      "  --qseq             query input files are in Illumina's qseq format\r\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\r\n",
      "  -r                 query input files are raw one-sequence-per-line\r\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\r\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\r\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\r\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\r\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\r\n",
      "  --phred33          qualities are Phred+33 (default)\r\n",
      "  --phred64          qualities are Phred+64\r\n",
      "  --int-quals        qualities encoded as space-delimited integers\r\n",
      "\r\n",
      " Presets:                 Same as:\r\n",
      "  For --end-to-end:\r\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\r\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\r\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\r\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\r\n",
      "\r\n",
      "  For --local:\r\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\r\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\r\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\r\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\r\n",
      "\r\n",
      " Alignment:\r\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\r\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\r\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\r\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\r\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\r\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\r\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\r\n",
      "  --nofw             do not align forward (original) version of read (off)\r\n",
      "  --norc             do not align reverse-complement version of read (off)\r\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\r\n",
      "                     scan for the optimal seeded alignments\r\n",
      "  --end-to-end       entire read must align; no clipping (on)\r\n",
      "   OR\r\n",
      "  --local            local alignment; ends might be soft clipped (off)\r\n",
      "\r\n",
      " Scoring:\r\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \r\n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\r\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\r\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\r\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\r\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\r\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\r\n",
      "\r\n",
      " Reporting:\r\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\r\n",
      "   OR\r\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\r\n",
      "   OR\r\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\r\n",
      "\r\n",
      " Effort:\r\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\r\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\r\n",
      "\r\n",
      " Paired-end:\r\n",
      "  -I/--minins <int>  minimum fragment length (0)\r\n",
      "  -X/--maxins <int>  maximum fragment length (500)\r\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\r\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\r\n",
      "  --no-discordant    suppress discordant alignments for paired reads\r\n",
      "  --no-dovetail      not concordant when mates extend past each other\r\n",
      "  --no-contain       not concordant when one mate alignment contains other\r\n",
      "  --no-overlap       not concordant when mates overlap at all\r\n",
      "\r\n",
      " Output:\r\n",
      "  -t/--time          print wall-clock time taken by search phases\r\n",
      "  --un <path>           write unpaired reads that didn't align to <path>\r\n",
      "  --al <path>           write unpaired reads that aligned at least once to <path>\r\n",
      "  --un-conc <path>      write pairs that didn't align concordantly to <path>\r\n",
      "  --al-conc <path>      write pairs that aligned concordantly at least once to <path>\r\n",
      "  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\r\n",
      "  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\r\n",
      "  --quiet            print nothing to stderr except serious errors\r\n",
      "  --met-file <path>  send metrics to file at <path> (off)\r\n",
      "  --met-stderr       send metrics to stderr (off)\r\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\r\n",
      "  --no-unal          suppress SAM records for unaligned reads\r\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\r\n",
      "  --no-sq            suppress @SQ header lines\r\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\r\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\r\n",
      "                     Note: @RG line only printed when --rg-id is set.\r\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\r\n",
      "\r\n",
      " Performance:\r\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\r\n",
      "  --reorder          force SAM output order to match order of input reads\r\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\r\n",
      "\r\n",
      " Other:\r\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\r\n",
      "  --seed <int>       seed for random number generator (0)\r\n",
      "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes\r\n",
      "  --version          print version information and quit\r\n",
      "  -h/--help          print this usage message\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bowtie2-2.2.9/bowtie2 -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"../data/Pg-70k-index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  ../data/Panopea_generosa-Scaff-70k.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 451843\n",
      "Using parameters --bmax 338883 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 338883 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 7; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 258195 (target: 338882)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (338883) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 299530 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 299531 for bucket 1\n",
      "Getting block 2 of 7\n",
      "  Reserving size (338883) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 148900 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 148901 for bucket 2\n",
      "Getting block 3 of 7\n",
      "  Reserving size (338883) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 323687 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 323688 for bucket 3\n",
      "Getting block 4 of 7\n",
      "  Reserving size (338883) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 284762 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 284763 for bucket 4\n",
      "Getting block 5 of 7\n",
      "  Reserving size (338883) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 208405 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 208406 for bucket 5\n",
      "Getting block 6 of 7\n",
      "  Reserving size (338883) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 324235 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 324236 for bucket 6\n",
      "Getting block 7 of 7\n",
      "  Reserving size (338883) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 217848 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 217849 for bucket 7\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 605139\n",
      "fchr[G]: 900679\n",
      "fchr[T]: 1194923\n",
      "fchr[$]: 1807373\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 4804841 bytes to primary EBWT file: ../data/Pg-70k-index.1.bt2\n",
      "Wrote 451848 bytes to secondary EBWT file: ../data/Pg-70k-index.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1807373\n",
      "    bwtLen: 1807374\n",
      "    sz: 451844\n",
      "    bwtSz: 451844\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 112961\n",
      "    offsSz: 451844\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 9414\n",
      "    numLines: 9414\n",
      "    ebwtTotLen: 602496\n",
      "    ebwtTotSz: 602496\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:02\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 451843\n",
      "Using parameters --bmax 338883 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 338883 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 258195 (target: 338882)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (338883) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 302407 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 302408 for bucket 1\n",
      "Getting block 2 of 7\n",
      "  Reserving size (338883) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 170619 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 170620 for bucket 2\n",
      "Getting block 3 of 7\n",
      "  Reserving size (338883) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 287319 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 287320 for bucket 3\n",
      "Getting block 4 of 7\n",
      "  Reserving size (338883) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 330746 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 330747 for bucket 4\n",
      "Getting block 5 of 7\n",
      "  Reserving size (338883) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 235349 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 235350 for bucket 5\n",
      "Getting block 6 of 7\n",
      "  Reserving size (338883) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 151345 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 151346 for bucket 6\n",
      "Getting block 7 of 7\n",
      "  Reserving size (338883) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 329582 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 329583 for bucket 7\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 605139\n",
      "fchr[G]: 900679\n",
      "fchr[T]: 1194923\n",
      "fchr[$]: 1807373\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 4804841 bytes to primary EBWT file: ../data/Pg-70k-index.rev.1.bt2\n",
      "Wrote 451848 bytes to secondary EBWT file: ../data/Pg-70k-index.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1807373\n",
      "    bwtLen: 1807374\n",
      "    sz: 451844\n",
      "    bwtSz: 451844\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 112961\n",
      "    offsSz: 451844\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 9414\n",
      "    numLines: 9414\n",
      "    ebwtTotLen: 602496\n",
      "    ebwtTotSz: 602496\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:01\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bowtie2-2.2.9/bowtie2-build \\\n",
    "../data/Panopea_generosa-Scaff-70k.fa \\\n",
    "../data/Pg-70k-index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "103213920 reads; of these:\r\n",
      "  103213920 (100.00%) were paired; of these:\r\n",
      "    102865097 (99.66%) aligned concordantly 0 times\r\n",
      "    261264 (0.25%) aligned concordantly exactly 1 time\r\n",
      "    87559 (0.08%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    102865097 pairs aligned concordantly 0 times; of these:\r\n",
      "      14413 (0.01%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    102850684 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      205701368 mates make up the pairs; of these:\r\n",
      "        205379458 (99.84%) aligned 0 times\r\n",
      "        205853 (0.10%) aligned exactly 1 time\r\n",
      "        116057 (0.06%) aligned >1 times\r\n",
      "0.51% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bowtie2-2.2.9/bowtie2 \\\n",
    "-x ../data/Pg-70k-index \\\n",
    "-1 /Volumes/Monarch/btea/Geoduck_v3/Geo_Pool_F_GGCTAC_L006_R1_001_val_2.fq \\\n",
    "-2 /Volumes/Monarch/btea/Geoduck_v3/Geo_Pool_F_GGCTAC_L006_R2_001_val_1.fq \\\n",
    "-S /Volumes/Monarch/course-fish546/Geo_Pool_F-Pg-70k.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HWI-D00743:26:C7C84ACXX:6:2316:21287:101065\t77\t*\t0\t0\t*\t*\t0\t0\tGTTATGGACCGTCTAAATCCTTACGACAAAGTTATGCGTAAAGCTGCTAAAGATCAGGAGGAACGTCGTAAGCGTGCCAGGTCAGGTGTTGTCGATGCTA\t@@@DADBDHDHDFEBHHGHECHEEGAE:CBD:BFBD?FBDAFHDGGHGGIHGHGHEHG@GF;?AB;2=9>AB>>53;AACB>>>CC:@:8ACCBB?D?CC\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21287:101065\t141\t*\t0\t0\t*\t*\t0\t0\tTTTACTTCTTGGCAGGTTTGCCTTTGGCGCCCTTAGCTGCTGCCTTCACTGGCTTCTTAGTGGCTGCTGCAGGACGTTTAGCATCGACAACACCTGACCT\t@@@DDDADFHHFDGIIHBEFHGGCGIIEFFGEGBHIGGE;BDGHBHHGE@=CFA;=@DEGDCG;AH;7?;));52;'5;5:(((,3,3&0(<02(939>:\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21317:101067\t77\t*\t0\t0\t*\t*\t0\t0\tNCCAGACACGTATCAAGTTGTCTGTGTATCCAGCAAAGAGTGTTTGTCCATCAGCTGACCAGGCCAAGCATGTACACTGTGGGGGCTCTGCTTTACAGCTG\t#1:ADDDDFFABFDBFGAEHACDHCHA9BDHFIFDFFGEG1B?DEGFGEIIIIEFCFFEFFIIF@@AEFEDE7???CAB@BDCBBB8<ABB@AAABBBB@A\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21317:101067\t141\t*\t0\t0\t*\t*\t0\t0\tGTGGTGGCACTATCAATGCTCTCACGTTCAGTCCAAACAGGTACTGGCTGTGTGCTGCAACTGGTCCCAGCATCAAGATCTGGGATCTTGAAGGCAAGGTT\t?1?DBDDDBBB?DFDFEHEHGCCGIIAFAA@FF@FECGGGHDDEBHII<GHHDGHCGGGDGGII=F@GAGACHCEEDB@@CDCCCCCCCCCCCC??2?B>:\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21267:101107\t77\t*\t0\t0\t*\t*\t0\t0\tATTTCATGAAAGGAATGTTATCCAATCCCAATGCTAGTGTCATGCATACACCGTTTTGATTGACATTGTTAACGTTAGCTTTATTTTGTTCTGTTCTGTTC\t@@@DFFFFDHHFHHGGGGFHIIHIIIIGIGIGIIICHHIDHIEHHGGHIEFHGDDGHIGGGHGDCHIIHEHIGIC?EHFDFCEBDBCEAECCDACCDCAC@\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21267:101107\t141\t*\t0\t0\t*\t*\t0\t0\tTAGGCACTATTTCACACATCGTCGGATNNCCATGGGAATCAATTCCGATACTCTCCATCCATCACCCGTGGGAAATTACGCTAGCCATGCATCGTTTTATT\t?@@DADDDBDHDHIIHIIGGIHGGIIE##11:BHFI@FAGGGGGGGHGFG@EGGHGHEHHF@DEEEECBBBBBCBCCCCBBB?BC?CCCCCDCB??BBCCC\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21256:101124\t77\t*\t0\t0\t*\t*\t0\t0\tCAGGATTTGAAATCGATGACCGCATCGCACCAGGATTCCGCTATTCTGTTCGATACGTTGGTAAAGATGAGTACCTGTTCGACGGAGAGGCAAGACGCTTG\t<<?DD3ADBD>C:C1:2ADE3FFAEEE=BDA@D7;9900?DA07=BCDCCCD2@AAC)7=>,77@A>>>>A>:;>;>;>><=<890;&8<<>(2395&5<>\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21256:101124\t141\t*\t0\t0\t*\t*\t0\t0\tTTAAAGTTGAGGCGATCCCCCTGGAANNNNNGGGGTCTTCCGTAACCGAGACCAATAGAATCCAAGCGTCTTGCCTCTCCGTCGAACAGGT\t=:++22,<A4A>A)70A<3??AAB;A#####00(00<ABB44.(5=.;'-59<>66>>>==??=3(55,5/;958==883098;;<208;8\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21274:101173\t77\t*\t0\t0\t*\t*\t0\t0\tGTGCATACCTAAGCCCGTCATTAGAGGATACCAATAATGAAACCCTGCAAATNTAGCAAATACTGCCCCTATACTTAAGACATAATGAAAATGTGCTACG\t@=@BBDDDFBDBFG3EGGEFI9CHEEF;?CFGDFBEFIFBFFFEIIIFFCDF#--;@C4@=DEEFEFEFCB@DDB@ACCCAAABABBBBBBBB@@B@ABB\tYT:Z:UP\r\n",
      "HWI-D00743:26:C7C84ACXX:6:2316:21274:101173\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNGCTTTAGGGTTTCTATTTCNNNNNNNNNNNGGGGGCTTAACTGGGATTGTTCTTTCTAGTGCTTCTTTAGATGTAATTTTACATGACACATATTATG\t####428;9>?@@@??<<??=??###########10:=??;>?;>????7;;?>>;?=??;??=????>?>>>?><=>?>>>>>>>>5=<=<=;;<====:\tYT:Z:UP\r\n"
     ]
    }
   ],
   "source": [
    "!tail /Volumes/Monarch/course-fish546/Geo_Pool_F-Pg-70k.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[samfaipath] build FASTA index...\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view \\\n",
    "-bT ../data/Panopea_generosa-Scaff-70k.fa \\\n",
    "/Volumes/Monarch/course-fish546/Geo_Pool_F-Pg-70k.sam \\\n",
    "> /Volumes/Monarch/course-fish546/Geo_Pool_F-Pg-70k.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001f�\b\u0004\u0000\u0000\u0000\u0000\u0000�\u0006\u0000BC\u0002\u0000j\u0002��1o�@\u0014�_`r%��Ć�.��}w>��\u0015)R\u001a\u0002A�\u0005YN⴦i\u001c",
      "����\f",
      "b��`@|\u0005F�\u0006##w�\u0010�w\u001bC$����������I��]���#��$\u0001��/��iU�\u0017�h����w��0\u001b���(x�nSR&�\r",
      "\u0019K\"\u0003�\u00183\u001b�Q�4�qjC�\r\n"
     ]
    }
   ],
   "source": [
    "!head -1 /Volumes/Monarch/course-fish546/Geo_Pool_F-Pg-70k.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/course-fish546_1DA47FD6.png\" alt=\"course-fish546_1DA47FD6.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Galaxy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Galaxy_1DA44C5F.png\" alt=\"Galaxy_1DA44C5F.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CoGe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/CoGe__My_Data_1DA4556B.png\" alt=\"CoGe__My_Data_1DA4556B.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/CoGe__My_Data_1DA45598.png\" alt=\"CoGe__My_Data_1DA45598.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<https://genomevolution.org/r/lslc>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
